pkg/maps/lbmap/lbmap.go

```diff
@@ -33,27 +34,31 @@ import (
 var log = logging.DefaultLogger.WithField(logfields.LogSubsys, "map-lb")
 
 var (
-	// MaxEntries contains the maximum number of entries that are allowed in cilium
-	// LB service, backend and affinity maps.
+	// MaxEntries contains the maximum number of entries that are allowed
+	// in Cilium LB service, backend and affinity maps.
 	MaxEntries = 65536
 )
 
 // LBBPFMap is an implementation of the LBMap interface.
 type LBBPFMap struct{}
 
+func New() *LBBPFMap {
+	return &LBBPFMap{}
+}
+
 // UpsertService inserts or updates the given service in a BPF map.
 //
 // The corresponding backend entries (identified with the given backendIDs)
 // have to exist before calling the function.
 //
 // The given prevBackendCount denotes a previous service backend entries count,
 // so that the function can remove obsolete ones.
-func (*LBBPFMap) UpsertService(
+func (lbmap *LBBPFMap) UpsertService(
 	svcID uint16, svcIP net.IP, svcPort uint16,
-	backendIDs []uint16, prevBackendCount int,
+	backends map[string]uint16, prevBackendCount int,
 	ipv6 bool, svcType loadbalancer.SVCType, svcLocal bool,
 	svcScope uint8, sessionAffinity bool, sessionAffinityTimeoutSec uint32,
-	checkSourceRange bool) error {
+	checkSourceRange, useMaglev bool) error {
```

Comment:
Might make sense to combine these in a `struct` parameter in a future PR.

Reply:
Yep, it's been in my TODO for quite a while. Added to the follow up issue: https://github.com/cilium/cilium/issues/13138.